python3Packages.name-that-hash: init at 1.10

eyJhb opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done
Build, tested both the binary as well as the binary.
Unsure if this is the correct way to get the binary named `nth`.

Also, had to patch `pyproject.toml`until a new version is released.
<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Related to https://github.com/NixOS/nixpkgs/issues/81418

eyJhb committed:
python3Packages.name-that-hash: init at 1.10

fabaff commented on pkgs/development/python-modules/name-that-hash/default.nix:
```diff
@@ -0,0 +1,41 @@
+}:
+
+buildPythonPackage rec {
+  pname = "nth";
```
The Python module should be named `name-that-hash`.

```suggestion
  pname = "name-that-hash";
```

Add it to `all-packages.nix` as `nth` with `toPythonApplication`.

fabaff commented on pkgs/development/python-modules/name-that-hash/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+  # TODO remove on next update which bumps rich
+  postPatch = ''
+    sed -i 's|rich = ">=9.9,<11.0"|rich = ">=9.9"|g' pyproject.toml
```
With `substituteInPlace` would do the same as `sed`.

RasmusRendal commented on pkgs/development/python-modules/name-that-hash/default.nix:
```diff
@@ -0,0 +1,41 @@
+  meta = with lib; {
+    description = "Don't know what type of hash it is? Name That Hash will name that hash type! Identify MD5, SHA256 and 300+ other hashes.";
+    homepage = "https://github.com/HashPals/Name-That-Hash";
+    license = with licenses; [ gpl3Only ];
```
Seems that the license is actually gpl3Plus.

eyJhb replied:
Fixed

fabaff commented on pkgs/development/python-modules/name-that-hash/default.nix:
```diff
@@ -0,0 +1,41 @@
+  meta = with lib; {
+    description = "Don't know what type of hash it is? Name That Hash will name that hash type! Identify MD5, SHA256 and 300+ other hashes.";
+    homepage = "https://github.com/HashPals/Name-That-Hash";
+    license = with licenses; [ gpl3Only ];
```
License is [GPLv3+](https://github.com/HashPals/Name-That-Hash/blob/main/pyproject.toml#L6).

```suggestion
    license = with licenses; [ gpl3Plus ];
```

fabaff reviewed (commented):
The upstream repo contains [tests](https://github.com/HashPals/Name-That-Hash/tree/main/tests). Could you check if they could be executed?

Add `pythonImportsCheck` as additional safeguard.

RasmusRendal reviewed (changes requested):
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64 linux
- [x] executables tested on x86_64 linux
- [x] all depending packages build

eyJhb commented:
> The upstream repo contains [tests](https://github.com/HashPals/Name-That-Hash/tree/main/tests). Could you check if they could be executed?
> 
> Add `pythonImportsCheck` as additional safeguard.

I have added it to check imports, should be OK now.
I am unsure if it runs the tests, however I couldn't find much on it.

I have fixed all the other comments now :)

fabaff reviewed (approved):
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/172296#issuecomment-11247155896)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-172296]$ ./results/nth/bin/nth 
Usage:  [OPTIONS]

  Name That Hash - Instantly name the type of any hash!
[...]
```
Looks good to me.
